tests/CreateConferenceFormTest.php

```diff
@@ -0,0 +1,223 @@
+            'ends_at' => 'potato',
+        ];
+
+        $form = CreateConferenceForm::fillOut($input, Factory::create('user'));
+        $form->complete();
+    }
+
+    /**
+     * @test
+     * @expectedException SaveMyProposals\Exceptions\ValidationException
+     */
+    public function conference_cfp_start_date_must_be_a_valid_date()
+    {
+        $input = [
+            'title' => 'AwesomeConf 2015',
+            'description' => 'The best conference in the world!',
+            'url' => 'http://example.com',
+            'cfp_starts_at' => 'potato',
+        ];
+
+        $form = CreateConferenceForm::fillOut($input, Factory::create('user'));
+        $form->complete();
+    }
+
+    /**
+     * @test
+     * @expectedException SaveMyProposals\Exceptions\ValidationException
+     */
+    public function conference_cfp_end_date_must_be_a_valid_date()
+    {
+        $input = [
+            'title' => 'AwesomeConf 2015',
+            'description' => 'The best conference in the world!',
+            'url' => 'http://example.com',
+            'cfp_ends_at' => 'potato',
+        ];
+
+        $form = CreateConferenceForm::fillOut($input, Factory::create('user'));
+        $form->complete();
+    }
+
+    /**
+     * @test
+     */
+    public function it_creates_a_conference_with_the_minimum_required_input()
+    {
+        $input = [
+            'title' => 'AwesomeConf 2015',
+            'description' => 'The best conference in the world!',
+            'url' => 'http://example.com',
+        ];
+
+        $form = CreateConferenceForm::fillOut($input, Factory::create('user'));
+        $form->complete();
+
+        $conference = Conference::first();
+        $this->assertEquals('AwesomeConf 2015', $conference->title);
+        $this->assertEquals('The best conference in the world!', $conference->description);
+        $this->assertEquals('http://example.com', $conference->url);
+    }
+
+    /**
+     * @test
+     */
+    public function conference_dates_are_saved_if_provided()
+    {
+        $input = [
+            'title' => 'AwesomeConf 2015',
+            'description' => 'The best conference in the world!',
+            'url' => 'http://example.com',
+            'starts_at' => '2015-02-01',
+            'ends_at' => '2015-02-04',
+            'cfp_starts_at' => '2015-01-15',
+            'cfp_ends_at' => '2015-01-18',
+        ];
+
+        $form = CreateConferenceForm::fillOut($input, Factory::create('user'));
+        $form->complete();
+
+        $conference = Conference::first();
+        $this->assertEquals(new DateTime('2015-02-01'), $conference->starts_at);
+        $this->assertEquals(new DateTime('2015-02-04'), $conference->ends_at);
+        $this->assertEquals(new DateTime('2015-01-15'), $conference->cfp_starts_at);
+        $this->assertEquals(new DateTime('2015-01-18'), $conference->cfp_ends_at);
+    }
+
+    /**
+     * @test
+     */
+    public function empty_dates_are_treated_as_null()
```

Comment:
Yeah like I mentioned above this is just for adding new conferences, I think what you're talking about matters for updating a conference but we still have that in the controller at the moment, nothing has been extracted there. Let's get that out next and write some tests around it then :)